recipes/default.rb

```diff
@@ -10,26 +10,14 @@
   compile_time false if respond_to?(:compile_time)
 end
 
-if platform_family? 'rhel'
-  yum_repository 'influxdb' do
-    description 'InfluxDB Repository - RHEL \$releasever'
-    baseurl node['influxdb']['upstream_repository']
-    gpgkey 'https://repos.influxdata.com/influxdb.key'
+influxdb_install 'influxdb' do
+  if node['kernel']['machine'].include? '64'
+    arch_type 'amd64'
+  elsif node['kernel']['machine'].include? '386'
+    arch_type 'i386'
+  else
+    Chef::Log.warning "I am unable to determine arch type for: #{node['kernel']['machine']}"
   end
```

Comment:
It makes more sense to me if you do this in the resource, not here in the recipe.